File: BlogAuthService/AuthService.Core/Helpers/PageInfo.cs
namespace AuthService.Core.Helpers
{
    using System;

    /// <summary>
    /// Page info.
    /// </summary>
    public class PageInfo
    {
        /// <summary>
        /// Gets or sets the page number.
        /// </summary>
        /// <value>
        /// The page number.
        /// </value>
        public int PageNumber { get; set; }

        /// <summary>
        /// Gets or sets the size of the page.
        /// </summary>
        /// <value>
        /// The size of the page.
        /// </value>
        public int PageSize { get; set; }

        /// <summary>
        /// Gets or sets the total items.
        /// </summary>
        /// <value>
        /// The total items.
        /// </value>
        public int TotalItems { get; set; }

        /// <summary>
        /// Gets the total pages.
        /// </summary>
        /// <value>
        /// The total pages.
        /// </value>
        public int TotalPages
        {
            get { return (int)Math.Ceiling((decimal)TotalItems / PageSize); }
        }
    }
}

File: Shared/Blog.FSharp.Core/Helpers/PageInfo.fs
namespace Blog.FSharp.Core.Helpers

open System

/// <summary>
/// Page information.
/// </summary>
type PageInfo() =
    /// <summary>
    /// Gets or sets the current page number.
    /// </summary>
    member val PageNumber: int = 0 with get, set

    /// <summary>
    /// Gets or sets the page size.
    /// </summary>
    member val PageSize: int = 0 with get, set

    /// <summary>
    /// Gets or sets the total number of items.
    /// </summary>
    member val TotalItems: int = 0 with get, set

    /// <summary>
    /// Gets the total number of pages.
    /// </summary>
    member this.TotalPages =
        if this.PageSize = 0 then 0
        else int (Math.Ceiling(decimal this.TotalItems / decimal this.PageSize))

